### PR TITLE
add `@typescript-eslint/no-floating-promises`

### PR DESCRIPTION
**Pull request by john-schmitz**
I am honestly not sure even if we should handle any errors comming from res.header() thats why I voided it

fixes https://github.com/trpc/trpc/issues/1577

### CODE REVIEW

**Review by KATT — COMMENTED** [review]
Not at a computer but have a look to see if any of the ❌ CI steps are obvious

**Review by skarab42 — APPROVED** [review]
Looks good :)

File: .tool-versions [hunk 1]
```diff
@@ -0,0 +1 @@
+nodejs 14.19.1
```
Comment (KATT) [inline .tool-versions:1]: Do `nvm` automatically pick this up

File: examples/next-prisma-starter/.eslintrc [hunk 2]
```diff
@@ -7,6 +7,7 @@
     "plugin:prettier/recommended"
   ],
   "parserOptions": {
+    "project": "tsconfig.json",
```
Comment (KATT) [inline examples/next-prisma-starter/.eslintrc:10]: This doesn't look related to this PR? We only need to change config at the root level of the project

```suggestion
```

File: packages/server/src/adapters/fastify/fastifyRequestHandler.ts [hunk 3]
```diff
@@ -67,13 +67,12 @@ export async function fastifyRequestHandler<
...
-
   for (const [key, value] of Object.entries(result.headers ?? {})) {
     if (typeof value === 'undefined') {
       continue;
     }
-    res.header(key, value);
-  }
 
-  res.send(result.body);
+    void res.header(key, value);
```
Comment (KATT) [inline packages/server/src/adapters/fastify/fastifyRequestHandler.ts:75]: Is `res.header` async here? Is that a fastify thing? cc @skarab42

File: packages/server/src/adapters/ws.ts [hunk 4]
```diff
@@ -285,7 +285,7 @@ export function applyWSSHandler<TRouter extends AnyRouter>(
         });
       }
     }
-    createContextAsync();
+    return createContextAsync();
```
Comment (KATT) [inline packages/server/src/adapters/ws.ts:288]:
```suggestion
    await createContextAsync();
```

File: packages/server/src/subscription.ts [hunk 5]
```diff
@@ -147,7 +147,7 @@ export function subscriptionPullFactory<TOutput>(opts: {
   }
 
   return new Subscription<TOutput>((emit) => {
-    _pull(emit);
+    _pull(emit).catch((err) => emit.error(getErrorFromUnknown(err)));
```
Comment (KATT) [inline packages/server/src/subscription.ts:150]: 👌 👌

**Comment by KATT** [comment]
@all-contributors add @john-schmitz for code

**Review by KATT — COMMENTED** [review]
The CI changes looks very much like they have to do with this PR. I tried re-running and it's fine on main

File: .tool-versions (hunk 1 above)
Comment (sachinraja) [reply .tool-versions]: If you're using asdf, you can have it [read from .nvmrc anyway](https://github.com/asdf-vm/asdf-nodejs#nvmrc-and-node-version-support). nvm does not read from `.tool-versions` at all though. Unfortunately, the ideal solution would be to have both.

Comment (KATT) [reply .tool-versions]: Let's use `.nvmrc` if that works for both 🤷

File: packages/server/src/adapters/fastify/fastifyRequestHandler.ts [hunk 6]
```diff
@@ -67,13 +67,12 @@ export async function fastifyRequestHandler<
...
+  }
+  return res.send(result.body);
```
Comment (KATT) [inline packages/server/src/adapters/fastify/fastifyRequestHandler.ts:77]:
```suggestion
    await res.send(result.body);
```

File: packages/server/src/adapters/fastify/fastifyRequestHandler.ts (hunk 3 above)
Comment (KATT) [reply packages/server/src/adapters/fastify/fastifyRequestHandler.ts:75]:
```suggestion
    await res.header(key, value);
```

**Review by KATT — CHANGES_REQUESTED** [review]
Some last tweaks then 🚀

**Comment by john-schmitz** [comment]
Sorry for the messed up configuration, did not realize that would break the examples.

But alas! It's fixed

**Review by KATT — APPROVED** [review]

**Comment by KATT** [comment]
@john-schmitz you have no GitHub sponsors setup, if you want to, set one up and I'll send you a thank you!

**Comment by john-schmitz** [comment]
> @john-schmitz you have no GitHub sponsors setup, if you want to, set one up and I'll send you a thank you!

I just requested for it! And thank you very much for your project, I contribuited first and foremost because I admire it.

**Comment by jlalmes** [comment]
Thanks for this - was meaning to get around to it at some point! 🚀
